stylesheets/commons/Fountain.less

```diff
@@ -0,0 +1,1756 @@
+	display: none;
+}
+
+/* Invert colors for dark theme */
+html.theme--dark .componentsinvert {
+	filter: invert( 1 );
+}
+
+/* Infobox Hero */
+.healthbar {
+	background: linear-gradient( to right, #286323, #7af03c );
+	color: #ffffff;
+	text-shadow: 1px 1px 2px #000000;
+	font-weight: bold;
+}
+
+.manabar {
+	background: linear-gradient( to right, #1056db, #73f5fe );
+	color: #ffffff;
+	text-shadow: 1px 1px 2px #000000;
+	font-weight: bold;
+}
+
+/* Infobox Cosmetic */
+.infobox-cosmetic-tradeable,
+.infobox-cosmetic-marketable {
+	background: radial-gradient( 50% 50% at 50% 50%, #03386f 0%, #030d1a 100% );
+	font-weight: bold;
+	color: #4898ed; // TODO: Need support for both true and false, this is the true color, false is #9f7669
+}
+
+.infobox-cosmetic-deletable {
+	background: radial-gradient( 50% 50% at 50% 50%, #262a30 0%, #000000 100% );
+	font-weight: bold;
+	color: #747b82; // TODO: Need support for both true and false, this is the false color, true is not defined
+}
+
+/* Text Colors */
+.text-color {
+	&-gold {
+		color: #daa520;
+	}
+}
+/* Template:Cf */
+a.cf {
+	font-weight: bold;
+	color: #ffffff;
+	text-shadow: 1px 1px 2px #000000;
+}
+
+/* Custom Diagrams */
+.diagram svg defs g path {
+	stroke: currentcolor;
+	fill: currentcolor;
+}
+
+.diagram svg defs g ellipse {
+	stroke: currentcolor;
+	fill: currentcolor;
+}
+
+/* Spellcard Ability Borders
+The smaller sizes are for sub-sections. */
+.target_passive,
+.target_aura {
+	border: 0;
+} /* Passive ability will have an overlay instead. */
+
+.target_no,
+.target_unit,
+.target_area,
+.target_ground,
+.target_point {
+	border-top: 6px solid #899499;
+	border-bottom: 6px solid #212f3d;
+	border-right: 6px solid #595959;
+	border-left: 6px solid #a7a7a7;
+}
+
+.target_toggle {
+	border: 6px solid #fc9303;
+}
+
+.target_altcast {
+	outline: 4px solid #f5f5f5;
+	outline-offset: -6px;
+	border-top: 6px solid #899499;
+	border-bottom: 6px solid #212f3d;
+	border-right: 6px solid #595959;
+	border-left: 6px solid #a7a7a7;
+}
+
+/* Spellcard Ability Borders (Autocast Animations) */
+*,
+*::before,
+*::after {
+	box-sizing: border-box;
+}
+
+@keyframes rotate {
+	100% {
+		transform: rotate( 1turn );
+	}
+}
+
+.target_autocast {
+	position: relative;
+	z-index: 0;
+	width: 80px;
+	height: 80px;
+	overflow: hidden;
+	padding: 6px;
+	outline: 1px solid #333333;
+	outline-offset: -4px;
+}
+
+.target_autocast::before {
+	content: "";
+	position: absolute;
+	z-index: -2;
+	left: -50%;
+	top: -50%;
+	width: 200%;
+	height: 200%;
+	background-color: transparent;
+	background-repeat: no-repeat;
+	background-image: linear-gradient( #000000, #fbb300 );
+	animation: rotate 1.2s cubic-bezier( 0.46, 0.03, 0.52, 0.96 ) infinite;
+}
+
+.target_autocast::after {
+	content: "";
+	position: absolute;
+	z-index: -1;
+	left: 6px;
+	top: 6px;
+	width: calc( 100% - 12px );
+	height: calc( 100% - 12px );
+	border-radius: 0;
+	display: flex;
+	justify-content: center;
+	align-items: center;
+}
```

Comment:
would write this as
```
.target_autocast {
  //add css here

  &::before {
    //add css here
  }

  &::after {
    //add css here
  }
}
```

Reply:
can either do
```
.target_autocast1::before,
.target_autocast::before {
```
and
```
.target_autocast1::after,
.target_autocast::after {
```

or do what you suggested here
```
.target_autocast {
  //add css here

  &::before {
    //add css here
  }

  &::after {
    //add css here
  }
}
```

(`target_autocast` and `target_autocast1` are a bit different sadly)

Reply:
alternatively could make one of the classes (the one used less) just overwrite some settings of the other one and adjust the usage (template/module) accordingly

Reply:
I would do
```
.target_autocast {
	width: 80px;
	height: 80px;
	padding: 6px;
}

.target_autocast1 {
	width: 40px;
	height: 40px;
	padding: 3px;
}

.target_autocast,
.target_autocast1 {
	//this is the css they both have in common
	position: relative;
	z-index: 0;
	overflow: hidden;
	outline: 1px solid #333333;
	outline-offset: -4px;

	&::before {
		content: "";
		position: absolute;
		z-index: -2;
		left: -50%;
		top: -50%;
		width: 200%;
		height: 200%;
		background-color: transparent;
		background-repeat: no-repeat;
		background-image: linear-gradient( #000000, #fbb300 );
		animation: rotate 1.2s cubic-bezier( 0.46, 0.03, 0.52, 0.96 ) infinite;
	}

	&::after {
		content: "";
		position: absolute;
		z-index: -1;
		left: 6px;
		top: 6px;
		width: calc( 100% - 12px );
		height: calc( 100% - 12px );
		border-radius: 0;
		display: flex;
		justify-content: center;
		align-items: center;
	}
}
```